File: libs/ruby-micromicro/test-one.rb
#!/usr/bin/env ruby

require 'bundler/setup'
require 'micromicro'
require 'json'

def print_usage
  puts 'Usage: test-one <input_file> <base_url>'
end

def process_html(file, baseUrl)
  out = MicroMicro.parse(IO.read(file), baseUrl)
  puts out.to_h.to_json
end

if ARGV[1].nil?
  print_usage
else
  process_html(ARGV[0], ARGV[1])
end

File: libs/python-robida/test-one.py
import mf2dom
import sys
import json

if len(sys.argv) < 3:
    print('usage: '+ sys.argv[0]+ ' <input_file> <base_url>')
    sys.exit(1)

name = sys.argv[1]
base = sys.argv[2]

with open(name,'r') as file:
    p = mf2dom.parse(file.read(), base_url=base)
    print(json.dumps(p))
